.github/workflows/reusable_testground_test.yaml

```diff
@@ -0,0 +1,113 @@
+        required: true
+        type: string
+        description: "TestGround Cluster Region"
+        default: "eu-west-1"
+      celestia_node_commit_sha:
+        required: false
+        type: string
+        description: "You can specify any commit that you want to use in the celestia-node repository"
+
+env:
+  REPO_TG: https://github.com/testground/testground.git
+  REPO_TEST_TG: https://github.com/celestiaorg/test-infra.git
+  ORG: celestiaorg
+  CELESTIA_NODE_REPO: celestia-node
+  BRANCH: main
+  TG_USER: ga-bot
+  TESTGROUND_URL: ${{ inputs.testground_url }}
+  TESTGROUND_REGION: ${{ inputs.testground_region }}
+  TEST_FILES_PATH: ${{ inputs.tests_files_path }}
+ 
+jobs:
+  testground:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - name: Clone the repo
+        run: |
+          git clone ${{ env.REPO_TG }}
+          git clone ${{ env.REPO_TEST_TG }}
+
+      - name: Build TestGround
+        run: |
+          export GOBIN=$(go env GOPATH)/bin
+          mkdir -p $GOBIN
+          cd testground
+          make goinstall
+
+      - name: Generate Config TestGround
+        run: |
+          mkdir -p ~/.config/testground/
+          cat <<EOF > ~/.config/testground/.env.toml
+          [aws]
+          region="${{ env.TESTGROUND_REGION }}"
+          [client]
+          endpoint = "${{ env.TESTGROUND_URL }}"
+          user="${{ env.TG_USER }}"
+          EOF
+          cat ~/.config/testground/.env.toml
+
+      # In the next step, we can verify if we need to use an specific commit 
+      # or we can use the one in the file.
+      - name: Update go.mod with the latest Celestia Node commit
+        run: |
+          COMMIT_SHA=$(echo '${{ inputs.celestia_node_commit_sha }}'| tr -d '"')
+          if [[ -z "$COMMIT_SHA" ]];then
+            echo "[INFO] There's no commit secified, so use the current version"
+          else
+            echo "[INFO] Let's use an specific commit SHA"
+            cd test-infra
+            # Get the latest commit SHA in the celestia-node repo
+            CELESTIA_NODE=$(curl -Ls https://api.github.com/repos/${ORG}/${CELESTIA_NODE_REPO}/commits/${BRANCH} | jq -c '.sha'| tr -d '"'| cut -c1-7)
+            echo $CELESTIA_NODE
+            # Replace the value in the go.mod file
+            sed -i "s|github.com/${ORG}/${CELESTIA_NODE_REPO}.*|github.com/${ORG}${CELESTIA_NODE_REPO} ${CELESTIA_NODE}|g" go.mod
+          fi
+
+      - name: Plan the execution
+        run: |
+          export GOBIN=$(go env GOPATH)/bin
+          cd test-infra
+          $GOBIN/testground plan import --from . --name celestia
+
+      - name: Start the TestGround Daemon
+        run: |
+          export GOBIN=$(go env GOPATH)/bin
+          cd test-infra
+          $GOBIN/testground plan import --from . --name celestia
+          $GOBIN/testground daemon &
+          # Let's give some seconds to the daemon service to start
+          sleep 3
```

Comment:
Is there a way to loop trough a check if the daemon is started? It might take longer than 3 seconds when GitHub has issues again.

Reply:
good one! 

I can add one check to the port, testground daemon is listening on port `8042`, we can see a message when it starts like:

```bash
Apr 27 13:11:06.377614  INFO    listen and serve        {"addr": "127.0.0.1:8042"}
Apr 27 13:11:06.377625  INFO    daemon listening        {"addr": "127.0.0.1:8042"}
```

so, I can add something like this:
```bash
until nc -z 127.0.0.1 8042;do
	sleep 1
	echo "waiting for the TestGround daemon to be up & running..."
done
echo "TestGround Daemon is ready to start receiving requests!"
```